src/features/amUI/users/UsersList.tsx

```diff
@@ -1,64 +1,81 @@
-import { Button, Heading, Tag, Pagination } from '@digdir/designsystemet-react';
+import React, { useState } from 'react';
+import { Button, Heading, Tag, Pagination, Search } from '@digdir/designsystemet-react';
 import classes from './UsersList.module.css';
 import { useTranslation } from 'react-i18next';
 import type { RightHolder } from '@/rtk/features/userInfo/userInfoApi';
-import { useGetRightHoldersQuery } from '@/rtk/features/userInfo/userInfoApi';
-import { getArrayPage, getTotalNumOfPages } from '@/resources/utils';
-
 import { ChevronDownIcon, ChevronUpIcon } from '@navikt/aksel-icons';
 
 import { ListItem } from '@/components/List/ListItem';
 import { List } from '@/components/List/List';
 import { UserIcon } from '@/components/UserIcon/UserIcon';
+import { useDebouncedFilteredRightHolders } from './useDebouncedFilteredRightHolders';
 
 export const UsersList = () => {
   const { t } = useTranslation();
 
   const [currentPage, setCurrentPage] = useState(1);
+  const [searchString, setSearchString] = useState<string>('');
   const pageSize = 10;
 
-  const { data: rightHolders } = useGetRightHoldersQuery();
+  const { pageEntries, numOfPages, searchResultLength } = useDebouncedFilteredRightHolders(
+    searchString,
+    currentPage,
+    pageSize,
+  );
 
-  const [pageEntrees, numOfPages] = useMemo(() => {
-    if (!rightHolders) {
-      return [[], 1];
-    }
-    const numPages = getTotalNumOfPages(rightHolders, pageSize);
-    return [getArrayPage(rightHolders, currentPage, pageSize), numPages];
-  }, [rightHolders, currentPage]);
+  const onSearch = (newSearchString: string) => {
+    setSearchString(newSearchString);
+    setCurrentPage(1); // reset current page when searching
+  };
 
   return (
     <div className={classes.usersList}>
+      <Search
+        className={classes.searchBar}
+        placeholder={t('users_page.user_search_placeholder')}
+        value={searchString}
+        onChange={(event) => onSearch(event.target.value)}
+        onClear={() => onSearch('')}
+      />
       <List
         compact
         heading={
           <Heading
             level={2}
-            size='lg'
+            size='md'
             spacing
           >
             {t('users_page.user_list_heading')}
           </Heading>
         }
       >
-        {pageEntrees.map((user) => (
+        {pageEntries.map((user) => (
           <UserListItem
             key={user.partyUuid}
             user={user}
           />
         ))}
       </List>
-      <Pagination
-        className={classes.pagination}
+      <Heading
+        level={2}
         size='sm'
-        hideLabels={true}
-        currentPage={currentPage}
-        totalPages={numOfPages}
-        onChange={(newPage) => setCurrentPage(newPage)}
-        nextLabel='Neste'
-        previousLabel='Forrige'
-      />
+        role='alert'
+        spacing
+      >
+        {searchResultLength === 0 ? t('users_page.user_no_search_result') : ''}
+      </Heading>
```

Comment:
Dynamically adding a DOM element that has aria-live or a role that has an implicit aria-live is typically not honored by the screen reader. Keeping it in the dom is a way to make sure the user gets the information we put in it. [link](https://stackoverflow.com/questions/76849643/best-practice-to-add-role-alert-in-react-conditional-rendering)

Reply:
True, but I get Wave errors (accessibility) on the empty heading.
Maybe the live region can instead be a div and then the content (the heading) can be rendered conditionally?
Or we can switch from a heading to a paragraph instead? Then it will no longer be an empty heading-error and as far as I know there is no rule against empty paragraphs

Reply:
Sure. A paragraph is a good option. I made it a heading as it is what NAV does, but it doesnt have to be a heading i think :)